### ISSUE TITLE
Inconsistent prediction outputs [BUG]

### ISSUE THREAD

**Issue by melisande-c** [CAREamics/careamics#156]
**Describe the bug**
The output dimensions of `CAREamics.predict` are different depending on whether tiling was used to compute the prediction or not. Additionally, if the no. of samples is greater than 1 then the output is a list, if there is only 1 sample then the prediction array is returned, not contained in a list.

If tiling is not used, the dimensions will always be SC(Z)YX; if the input did not have the C dim it is set to 1, and S is always 1 (because predictions are returned as a list i.e. a list of length S (unless S=1) of arrays with shape (1, C, Y, X). If tiling is used then the S dimension is always removed, and if C is 1, then it is removed.

**To Reproduce**
Code snippet allowing reproducing the behaviour:
***Tiled vs non-tiled***
```python
import numpy as np

from careamics import CAREamist
from careamics.config import create_n2v_configuration

config = create_n2v_configuration(
    experiment_name="PredBatchingTest", 
    data_type="array",
    axes="SYX",
    patch_size=[8, 8],
    batch_size=1,
    num_epochs=1,
    n_channels=1,
    )
image = np.random.random((1, 32, 32))

engine = CAREamist(source=config)
engine.train(train_source=image)
pred_tiled = engine.predict(source=image, tile_size=(8, 8), tile_overlap=(2,2))
pred = engine.predict(source=image)

print("Tiled shape: ", pred_tiled.shape)
print("Non-tiled shape: ", pred.shape)
```
```
>>> ...
>>> Tiled shape:  (32, 32)
>>> Non-tiled shape:  (1, 1, 32, 32)
```
***List output***
```
import numpy as np

from careamics import CAREamist
from careamics.config import create_n2v_configuration

config = create_n2v_configuration(
    experiment_name="PredBatchingTest", 
    data_type="array",
    axes="SYX",
    patch_size=[8, 8],
    batch_size=1,
    num_epochs=1,
    n_channels=1,
    )
image = np.random.random((2, 32, 32))

engine = CAREamist(source=config)
engine.train(train_source=image)
pred = engine.predict(source=image)

print("Output type: ", type(pred))
print("Output length: ", len(pred))
print("Prediction shape: ", pred[0].shape)
```
```
>>> ...
>>> Output type:  <class 'list'>
>>> Output length:  2
>>> Prediction shape:  (1, 1, 32, 32)
```

**Expected behavior**
Prediction outputs should be consistent. 

I suggest that the outputs should always be contained in a list, even if there is only 1 sample. I would not concatenate the samples because there is a possibility that, when predicting on a set of files, the samples have different YX dimensions.

I suggest that the samples should always have the channel dimension added, even if the input did not have channels; but the S dimension should be removed since it will always be 1, since the channels are returned in a list.

**Causes**
The function `stitch_prediction` assumes there is no S dim, so it is removed from the inputs before calling the function and it is never added back. I am not sure how the C dimension is removed.

**Comment by melisande-c** [comment]
In my description of the issue above, I suggested that the channel dimension should always be present – for consistency. However I have run into some problems with singleton dimensions being disallowed.

The channel dimension is removed from the tiled output in the `stitched_prediction_single` function when the `predicted_image` array is created from the attribute `TileInformation.array_shape` attribute (see code ref).
https://github.com/CAREamics/careamics/blob/2764100c599cf911999d9f30be7cf76b87fd917b/src/careamics/prediction_utils/stitch_prediction.py#L78-L79

Singleton dimensions in the `array_shape` attribute are specifically disallowed in the `TileInformation` Pydantic model.
https://github.com/CAREamics/careamics/blob/2764100c599cf911999d9f30be7cf76b87fd917b/src/careamics/config/tile_information.py#L27-L29

So the choice is to relax the constraints on singleton dimensions; or to simply remove singleton dimensions from the non-tiled predictions as well. The third more slightly complicated choice is to try to match the dimensions of the input and output.

In relation to the discussion in #159, in my opinioin singleton dimensions should be allowed as input. If we go for the option where singleton dimensions are always removed, then clearly the prediction output in this case would not have the same singleton dimension. If this is clearly documented, I think it should be ok.
